apps/heft/src/plugins/TypescriptPlugin/TypescriptBuilder.ts

```diff
@@ -0,0 +1,1029 @@
+      options: tsconfig.options,
+      projectReferences: tsconfig.projectReferences,
+      host: compilerHost,
+      configFileParsingDiagnostics: ts.getConfigFileParsingDiagnostics(tsconfig)
+    });
+
+    this._terminal.writeVerboseLine(
+      `I/O Read: ${ts.performance.getDuration('I/O Read')}ms (${ts.performance.getCount(
+        'beforeIORead'
+      )} files)`
+    );
+    this._terminal.writeVerboseLine(
+      `Parse: ${ts.performance.getDuration('Parse')}ms (${ts.performance.getCount('beforeParse')} files)`
+    );
+    this._terminal.writeVerboseLine(
+      `Program (includes Read + Parse): ${ts.performance.getDuration('Program')}ms`
+    );
+    //#endregion
+
+    //#region DIAGNOSTICS
+    const { duration: diagnosticsDurationMs, diagnostics } = measureTsPerformance('Diagnostics', () => {
+      const rawDiagnostics: Typescript.Diagnostic[] = [
+        ...tsProgram.getConfigFileParsingDiagnostics(),
+        ...tsProgram.getOptionsDiagnostics(),
+        ...tsProgram.getSyntacticDiagnostics(),
+        ...tsProgram.getGlobalDiagnostics(),
+        ...tsProgram.getSemanticDiagnostics()
+      ];
+      const _diagnostics: ReadonlyArray<Typescript.Diagnostic> = ts.sortAndDeduplicateDiagnostics(
+        rawDiagnostics
+      );
+      return { diagnostics: _diagnostics };
+    });
+    this._terminal.writeVerboseLine(`Diagnostics: ${diagnosticsDurationMs}ms`);
+    //#endregion
+
+    //#region EMIT
+    const filesToWrite: IFileToWrite[] = [];
+    const writeFileCallback: Typescript.WriteFileCallback = (filePath: string, data: string) => {
+      filesToWrite.push({ filePath, data });
+    };
+
+    const emitResult: IExtendedEmitResult = this._emit(ts, tsconfig, tsProgram, writeFileCallback);
+    //#endregion
+
+    this._terminal.writeVerboseLine(`Bind: ${ts.performance.getDuration('Bind')}ms`);
+    this._terminal.writeVerboseLine(`Check: ${ts.performance.getDuration('Check')}ms`);
+    this._terminal.writeVerboseLine(
+      `Transform: ${ts.performance.getDuration('transformTime')}ms ` +
+        `(${ts.performance.getCount('beforeTransform')} files)`
+    );
+    this._terminal.writeVerboseLine(
+      `Print: ${ts.performance.getDuration('printTime')}ms ` +
+        `(${ts.performance.getCount('beforePrint')} files) (Includes Transform)`
+    );
+    this._terminal.writeVerboseLine(`Emit: ${ts.performance.getDuration('Emit')}ms (Includes Print)`);
+
+    //#region WRITE
+    const writePromise: Promise<{ duration: number }> = measureTsPerformanceAsync('Write', () =>
+      Async.forEachLimitAsync(filesToWrite, ASYNC_LIMIT, async ({ filePath, data }) =>
+        this._fileSystem.writeFile(filePath, data, { ensureFolderExists: true })
+      )
+    );
+    //#endregion
+
+    const typescriptFilenames: Set<string> = new Set(tsconfig.fileNames);
+
+    //#region TSLINT
+    let tslintResult: Tslint.LintResult | undefined = undefined;
+    if (tslint) {
+      tslintResult = await this._runTslintAsync({
```

Comment:
This file is getting kind of large. Would it be easy to split `_runTslintAsync()` into a separate file? That way ESLint and TSLint won't both be mixed up with the other unrelated code in TypeScriptBuilder.ts

Reply:
Hmm that'll be kind of a PITA. Let's consider refactoring when ESLint support is added.